akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -185,6 +185,12 @@ object Actor {
       new BehaviorImpl.ImmutableBehavior(onMessage, onSignal)
   }
 
+  /**
+   * Construct an immutable actor behavior from a partial message handler which treats undefined messages as unhandled.
+   */
+  def immutableOrUnhandled[T](onMessage: PartialFunction[(ActorContext[T], T), Behavior[T]]): Immutable[T] =
+    Actor.immutable[T] { (ctx, t) ⇒ onMessage.applyOrElse((ctx, t), (_: (ActorContext[T], T)) ⇒ Actor.unhandled[T]) }
```

Comment:
I skimmed the ticket but it seems a more compositional style wasn't really considered?
I would want to avoid having "thisOrThat" methods in the most core APIs, it makes it look (and does) as if the behaviour composition is breaking down and we "have to" add such methods that do "Or".

```
Actor.immutable(Actor.partialReceive { 
    case "hello" => 
  }
}
```

Reply:
There's no `Actor.partialReceive`. So you are suggesting to introduce that?

I think this is a great idea! Composition over bullshit ;-)

In Typed we don't use "receive" (except for `MutableBehavior` which I tend to ignore), do we? So I suggest to rename this new method `Actor.partial`.

Reply:
Yeah, proposing we add that, with the impl that you have here in the `immutableOrUnhandled` basically :) `Actor.partial` also sounds good to me :)

Reply:
@ktoso I don't fully understand how to compose (wrap) behaviors. `supervise`, `withTimers`, etc. all take a `Behavior[T]` or factories for a `Behavior[T]` as arguments. But `immutable` takes a function (message handler). So `partial` obviously can't return a `Behavior[T]`. What am I missing?

Reply:
wasn't that exactly what I initially proposed in the ticket?
```
def onMessageOrIgnore[T](onMessage: PartialFunction[(ActorContext[T], T), Behavior[T]]): (ActorContext[T], T) => Behavior[T]
```

Then I vote for that, again.

Reply:
Hmm, I see, this might require adding some overload to immutable perhaps hm hm...

Reply:
Put another way: behaviors can be wrapped, but `immutable` is sort of an innermost behavior and it's message handler can't.

Reply:
and the name `Actor.partial` is good

Reply:
I'll send in a PR showing the general idea so you can take it from there if you want @hseeberger (have it ready)

Reply:
+1